hub/core/chunk/chunk_compressed_chunk.py

```diff
@@ -1,123 +1,154 @@
 
             if isinstance(serialized_sample, SampleTiles):
-                incoming_samples[i] = serialized_sample
+                incoming_samples[i] = serialized_sample  # type: ignore
                 if self.is_empty:
-                    self.write_tile(serialized_sample)
-                    num_samples += 0.5
+                    self.write_tile(serialized_sample, skip_bytes=True)
+                    num_samples += 0.5  # type: ignore
                     tile = serialized_sample.yield_uncompressed_tile()
-                    self._decompressed_bytes = tile.tobytes()
+                    self.decompressed_bytes = tile.tobytes()
+                    self._changed = True
                 break
-            else:
-                sample_nbytes = len(serialized_sample)
-                buffer += serialized_sample
-                compressed_bytes = compress_bytes(buffer, self.compression)
-                if self.is_empty or len(compressed_bytes) < self.min_chunk_size:
-                    self.data_bytes = compressed_bytes
-                    self.register_in_meta_and_headers(sample_nbytes, shape)
-                    num_samples += 1
-                    self._decompressed_bytes = buffer
-                else:
+            sample_nbytes = len(serialized_sample)
+
+            recompressed = False  # This flag helps avoid double concatenation
+            if (
+                len(self.decompressed_bytes) + sample_nbytes  # type: ignore
+            ) * self._compression_ratio > self.min_chunk_size:
+                new_decompressed = self.decompressed_bytes + serialized_sample  # type: ignore
+                compressed_bytes = compress_bytes(
+                    new_decompressed, compression=self.compression
+                )
+                if len(compressed_bytes) > self.min_chunk_size:
                     break
+                recompressed = True
+                self.decompressed_bytes = new_decompressed
+                self._compression_ratio /= 2
+                self._data_bytes = compressed_bytes
+                self._changed = False
+            if not recompressed:
+                self.decompressed_bytes += serialized_sample  # type: ignore
+            self._changed = True
+            self.register_in_meta_and_headers(sample_nbytes, shape)
+            num_samples += 1
         return num_samples
 
-    def extend_if_has_space_img_compression(self, incoming_samples: List[InputSample]):
-        num_samples: float = 0
-        buffer_list = self.decompressed_samples if self.data_bytes else []
-
+    def extend_if_has_space_image_compression(
+        self, incoming_samples: Union[Sequence[InputSample], np.ndarray]
+    ):
+        num_samples = 0
+        num_decompressed_bytes = sum(
+            x.nbytes for x in self.decompressed_samples  # type: ignore
+        )
         for i, incoming_sample in enumerate(incoming_samples):
+            if isinstance(incoming_sample, bytes):
```

Comment:
we can remove this, incoming sample will never be bytes, I think I had kept this earlier as we were hoping to chain compressions at some point